cmd/preview/list.go

```diff
@@ -72,16 +86,72 @@ func executeListPreviews(ctx context.Context, opts ListFlags) error {
 		}
 		return fmt.Errorf("failed to get preview environments: %s", err)
 	}
-	w := tabwriter.NewWriter(os.Stdout, 1, 1, 2, ' ', 0)
-	fmt.Fprintf(w, "Name\tScope\tSleeping\tLabels\n")
+	previewListOutput, err := getPreviewOutput(ctx, opts)
+	switch opts.output {
+	case "json":
+		if err != nil {
+			return err
+		}
+		bytes, err := json.MarshalIndent(previewListOutput, "", " ")
+		if err != nil {
+			return err
+		}
+		oktetoLog.Println(string(bytes))
+	case "yaml":
+		bytes, err := yaml.Marshal(previewListOutput)
+		if err != nil {
+			return err
+		}
+		oktetoLog.Println(string(bytes))
+	default:
+		w := tabwriter.NewWriter(os.Stdout, 1, 1, 2, ' ', 0)
+		fmt.Fprintf(w, "Name\tScope\tSleeping\tLabels\n")
+		for _, preview := range previewList {
+			previewLabels := "-"
+			if len(preview.PreviewLabels) > 0 {
+				previewLabels = strings.Join(preview.PreviewLabels, ", ")
+			}
+			fmt.Fprintf(w, "%s\t%s\t%v\t%s\n", preview.ID, preview.Scope, preview.Sleeping, previewLabels)
+		}
+		w.Flush()
+	}
+	return nil
+}
+
+func getPreviewOutput(ctx context.Context, opts ListFlags) ([]PreviewOutput, error) {
+	var previewSlice []PreviewOutput
+	oktetoClient, err := okteto.NewOktetoClient()
+	if err != nil {
+		return nil, err
+	}
+	previewList, err := oktetoClient.Previews().List(ctx, opts.labels)
+	if err != nil {
+		if uErr, ok := err.(oktetoErrors.UserError); ok {
+			return nil, uErr
+		}
+		return nil, fmt.Errorf("failed to get preview environments: %s", err)
+	}
 	for _, preview := range previewList {
 		previewLabels := "-"
 		if len(preview.PreviewLabels) > 0 {
 			previewLabels = strings.Join(preview.PreviewLabels, ", ")
 		}
-		fmt.Fprintf(w, "%s\t%s\t%v\t%s\n", preview.ID, preview.Scope, preview.Sleeping, previewLabels)
+		previewOutput := PreviewOutput{
+			Name:     preview.ID,
+			Scope:    preview.Scope,
+			Sleeping: preview.Sleeping,
+			Labels:   previewLabels,
+		}
+		previewSlice = append(previewSlice, previewOutput)
 	}
+	return previewSlice, nil
+}
 
-	w.Flush()
-	return nil
+func ValidateOutput(output string) error {
```

Comment:
```suggestion
func ValidateOutputFlag(output string) error {
```

Just to improve readability of this function, and avoiding the reader to think it's about validating the actual output